chore: Use Bun for package management

aklinker1 opened the pull request:
This closes #1643.

- [x] Update scripts
- [ ] Migrate E2E tests to use bun

aklinker1 committed:
fix: Replace all pnpm calls with bun

aklinker1 committed:
Cleanup

aklinker1 committed:
fix netlify

aklinker1 committed:
Fix api ref

aklinker1 committed:
Fix npm package test

aklinker1 committed:
cleanup

aklinker1 committed:
fix

aklinker1 committed:
fix pnpx ref

aklinker1 committed:
Revert

okineadev commented:
After merging this PR, we can then migrate all `vitest` tests to `bun:test`

It's not urgent, everything is drop-in replacement there except mocks, in Bun it's a bit specific

okineadev commented on .gitattributes:
```diff
@@ -6,6 +6,7 @@
 # Mark lock files as generated to avoid diffing
 pnpm-lock.yaml linguist-generated
 package-lock.json linguist-generated
```
```suggestion
```

aklinker1 replied:
They are used in test fixures, that's why they're there.

okineadev commented on .prettierignore:
```diff
@@ -4,5 +4,6 @@ dist
 .wxt
 docs/.vitepress/cache
 pnpm-lock.yaml
```
```suggestion
```

aklinker1 replied:
Used in test fixture

okineadev commented on .gitattributes:
```diff
@@ -6,6 +6,7 @@
 package-lock.json linguist-generated
+bun.lock linguist-generated
 bun.lockb linguist-generated
 yarn.lock linguist-generated
```
```suggestion
```

aklinker1 replied:
Used in test fixture.

okineadev commented on .github/actions/setup/action.yml:
```diff
@@ -1,31 +1,27 @@
     default: ''
     type: string
-    description: Additional args to append to "pnpm install"
+    description: Additional args to append to "bun i"
```
```suggestion
    description: Additional args to append to "bun install"
```

okineadev commented on .github/actions/setup/action.yml:
```diff
@@ -1,31 +1,27 @@
     default: 'true'
     type: boolean
-    description: Whether or not to run 'pnpm install'
+    description: Whether or not to run 'bun i'
```
```suggestion
    description: Whether or not to run 'bun install'
```

aklinker1 committed:
use bun run

aklinker1 commented on .gitattributes:
```diff
@@ -6,6 +6,7 @@
 pnpm-lock.yaml linguist-generated
 package-lock.json linguist-generated
+bun.lock linguist-generated
 bun.lockb linguist-generated
```
There's still a test fixture containing a binary lockfile, I'll try and remove this one when I update E2E tests to use bun.

okineadev replied:
> There's still a test fixture containing a binary lockfile, I'll try and remove this one when I update E2E tests to use bun.

Okay

aklinker1 commented on .github/workflows/release.yml:
```diff
@@ -52,18 +52,18 @@ jobs:
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_AUTH_TOKEN }}" > ~/.npmrc
-          pnpm build
-          pnpm publish
+          bun run build
+          bun publish
```
WXT doesn't have provenance right now, so I'll do this in a separate PR.

okineadev replied:
> WXT doesn't have provenance right now, so I'll do this in a separate PR.
> 
> #1654

Ok

aklinker1 committed:
Update packages/browser/scripts/generate.ts

Co-authored-by: Okinea Dev <hi@okinea.dev>

aklinker1 commented on scripts/benchmarks/browser-startup.sh:
```diff
@@ -16,17 +16,17 @@ N=20
     git apply scripts/benchmarks/browser-startup.patch
-    pnpm i --ignore-scripts
-    pnpm -r --filter wxt build
+    bunx i --ignore-scripts
+    bunx --filter wxt build
```
Woops

aklinker1 committed:
Cleanup

aklinker1 commented:
> After merging this PR, we can then migrate all `vitest` tests to `bun:test`
> 
> It's not urgent, everything is drop-in replacement there except mocks, in Bun it's a bit specific

Yeah, if we decide to migrate tests, we'll do that in a separate PR.

aklinker1 committed:
Fix formatting

okineadev commented:
> [`1e81f7c` (#1649)](https://github.com/wxt-dev/wxt/pull/1649/commits/1e81f7c230f378aaa43436b47a8b214f21cbccf4)

Please add `Co-Authored-By: Okinea Dev <hi@okinea.dev>` next time 🙏

aklinker1 committed:
fix esbuild versions

okineadev commented:
Errors in tests due to lack of pnpm?

It seems that Node.js and pnpm should be added to the test workflows

aklinker1 commented:
> Errors in tests due to lack of pnpm?
> 
> It seems that Node.js and pnpm should be added to the test workflows

Yeah, boarding a plane rn and traveling all day today, so this will have to wait until tomorrow.

aklinker1 commented on .github/workflows/update-browser-package.yml:
```diff
@@ -43,4 +43,4 @@ jobs:
         run: |
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_AUTH_TOKEN }}" > ~/.npmrc
-          pnpm publish
+          bun publish
```
Resolved in https://github.com/wxt-dev/wxt/pull/2009

aklinker1 commented:
@okineadev I'll make sure to add `Co-Authored-By: Okinea Dev <hi@okinea.dev>` to #2009 when I merge it, I've double checked that I've addressed all your feedback from this PR on that one as well.
